### DISCUSSION TITLE
`x typescript` creates types with errors

### DISCUSSION THREAD

**Discussion by matthew-dean** [ServiceStack/Discuss#62] [Q&A]
In our code-base, we have C# definitions like:
```csharp
public class ProductMetadata : ProductAPI.Contracts.V1.DTOs.ProductMetadata
```

When running `x typescript`, this is output as:
```ts
export class ProductMetadata extends ProductMetadata
```
...which of course is invalid. Is there way to force type generation to actually be valid? How do people handle this common scenario?

**Comment by matthew-dean** [comment]
Btw, where is the source code for the tool that generates this? The ServiceStack repo organization is really confusing, so I wasn't able to find it.

**Comment by mythz** [comment]
All [DTO Type Names need to be unique](https://docs.servicestack.net/add-servicestack-reference#limitations) to use Add ServiceStack Reference so you can't have duplicate type names.

The `x` and `app` dotnet tools are in https://github.com/ServiceStack/dotnet-app but all DTO Generation happens inside ServiceStack, e.g. TypeScript generation happens in 
https://github.com/ServiceStack/ServiceStack/blob/main/ServiceStack/src/ServiceStack/NativeTypes/TypeScript/TypeScriptGenerator.cs which generates TypeScript DTOs at the `/types/typescript` endpoint.
